### PR TITLE
MINIFICPP-2067 Remove OpenSSL system dependency from MQTT

### PR DESCRIPTION
**Pull request by lordgamez**
https://issues.apache.org/jira/browse/MINIFICPP-2067

-----------------
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by szaszm** [comment]
This fixed the windows cmake error for me. The build didn't finish yet, but it used to fail in the configure step.

File: thirdparty/paho-mqtt/cmake-openssl.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
++++ b/src/CMakeLists.txt
+@@ -187,7 +187,7 @@ INSTALL(FILES MQTTAsync.h MQTTClient.h MQTTClientPersistence.h MQTTProperties.h
+ 
+ IF (PAHO_WITH_SSL)
+     SET(OPENSSL_ROOT_DIR "" CACHE PATH "Directory containing OpenSSL libraries and includes")
+-    find_package(OpenSSL REQUIRED)
++    # find_package(OpenSSL REQUIRED)
+ 
+     IF (PAHO_BUILD_SHARED)
+         ## common compilation for libpaho-mqtt3cs and libpaho-mqtt3as
```
[the remark this answers is not in the document]
Comment (lordgamez) [reply thirdparty/paho-mqtt/cmake-openssl.patch]: I checked the CMake logs and you are right it finds the right package without it, I removed it in 100d83d9751c09383ad6a46ce3bb87a9a5239e09

**Comment by lordgamez** [comment]
The only failed test is from the `ExpressionLanguageTests` suite from the line:
```
CHECK( LogTestController::getInstance().contains("reverseDnsLookup timed out", 0ms) )
```
It looks to be a flaky test from the Reverse DNS lookup feature.

**Comment by szaszm** [comment]
I've asked @martinzink to have a look at that test. I wouldn't block this PR because of that failure.

**Review by adamdebreceni — APPROVED** [review]

**Review by fgerlits — APPROVED** [review]

**Pull request closed without merging**